### ISSUE TITLE
ProteusVX VST and LMMS only noisy fragmented sound

### ISSUE THREAD

**Issue by lmmsuser1001** [LMMS/lmms#2440]
Hello,

I am using LMMS version 1.1.3 with windows XP and the free vst plugin ProteusVX from:

http://www.creative.com/emu/proteusvx/

Unfortunately, I only get a noisy fragmented sound with this plugin.

The problem is also discussed here, but without a solution:

https://lmms.io/forum/viewtopic.php?t=3556

With other DAWs the plugin works fine or when I start ProteusVX directly as executable application.

Do you have any hints?

**Comment by musikBear** [comment]
> The problem is also discussed here, but without a solution:

And there is yet no new release, so the 'solution' has not changed
Suggest close

**Comment by Umcaruje** [comment]
>And there is yet no new release, so the 'solution' has not changed
>Suggest close

Your answer to the OP's question on the forums is painfully wrong. We don't use the steinberg SDK, so our VST support has nothing to so with "versions of steinberg 'kits'". We use VeSTige, which is a reverse-engineered solution, but as that, it has some bugs and is not 100% compatible with all VST's.

You stated that the OP should "Try again after next release", but that's also wrong, since nobody from the dev team will know that that particular plugin doesn't work, and it won't be fixed. If you suspect that the changes in the VST support from the master branch might have fixed this plugin, you should offer the OP one of the current builds of the master branch, so he can test it.

In conclusion, your post was not a solution at all, so you please don't suggest that this issue should be closed because of your incompetence to present a proper solution to a user on the forums.

**Comment by musikBear** [comment]
> you should offer the OP one of the current builds of the master branch, so he can test it.

I was under the impression that no builds were stable. Must be my blatant incompetence once more. Luckily im not unpleasant.

**Comment by tresf** [comment]
> I was under the impression that no builds were stable. Must be my blatant incompetence once more. Luckily im not unpleasant.

You're simply missing the point.  It was YOU that proposed a new version **might** fix the problem, so it is on **you** as a veteran contributor to help the OP test that.

Challenging @Umcaruje about the stability of the master branch is a cop-out to something that you proposed.  Don't turn this on him, that's not a nice thing to do. :smile:

**Comment by tresf** [comment]
I just installed this plugin in Windows XP and experienced the choppy sound in the stand-alone version.

 1. Download ProteusVX VST here: http://origin.creative.com/emu/files/ProteusVX.zip
 2. Extract the ZIP
 3. Run installer via `ProtVX\Setup.exe`
 4. Run ProteusVX from shortcut
 5. Set audio to `DirectSound` via Options, Preferences
 6. Selected `JP-Moog 4` from list of presets
 7. Enable onscreen keyboard via View, Keyboard (F9)
 8. Play notes are echoed and choppy (possibly as described).
 9. Continued playing for a while and the audio seems to eventually correct itself.

Next, I opened a new LMMS project using version `1.1.3` from our downloads page

 1. Click File Browser on left, My Computer
 2. Navigate to `C:\Program Files\Steinberg\VSTPlugins\E-MU\ProteusVX.dll`
 3. Drag into new instrument track
 4. Can't find `JP-Moog 4`, so I re-opened it from the desktop shortcut and exported it to the desktop.
 6. Enabled onscreen keyboard via View, Keyboard (F9).  Playing was chopping at first (like desktop application) and cleared up after playing a few notes.
 7. Created an LMMS piano roll using new instrument
 8. Plays fine.  Listen [here `proteus.ogg`](https://drive.google.com/file/d/0B4PpvIwHd1U0SjJSME11NEROcGc/view?usp=sharing).

Worth mentioning, I tried tweaking the settings, such as setting the buffer size to 256 and the CPU cap to 80% via the Proteus preferences.  I'm not sure if this helped, but after changing settings the stuttering seemed to get better.

In any case, this problem seems to be directly related to the plugin (possibly the CPU cap) since this was reproducible whether or not LMMS was running.

Closing as invalid, since this bug has to do with the VST plugin itself.  If you can find evidence otherwise, let us know and we can reopen.

**Comment by lmmsuser1001** [comment]
Thank you very much for your great support!

I am using the 32-Bit version of LMMS and ProteusVX. Possibly, it is a 
32-Bit problem? Did you work with the 32-Bit version?

If I use the stand-alone version of ProteusVX, I hear a sound that is 
similiar to the sound in the sound file you sent: |proteus.ogg| 
<https://drive.google.com/file/d/0B4PpvIwHd1U0SjJSME11NEROcGc/view?usp=sharing>.

But If I start LMMS with the VST-Plugin, I hear a completely noisy sound 
(attached wave file). The waveform is regularly distorted. The attached 
wave file was exported from LMMS. So, I think this is not a problem that 
is influenced by the selection of the output buffer size.

Furthermore, I recognized that the visualization of the master output is 
always cut behind the center of the display when playing a note with 
ProteusVX in LMMS (see screenhot).

Thank you very much.


Am 09.11.2015 um 20:47 schrieb Tres Finocchiaro:

**Comment by lmmsuser1001** [comment]
Attached screenshot of the comment above:

![proteusvx_lmms](https://cloud.githubusercontent.com/assets/12001122/11049380/1cec03ce-873e-11e5-9645-fa9e08af37df.JPG)

**Comment by tresf** [comment]
> I am using the 32-Bit version of LMMS and ProteusVX. Possibly, it is a 32-Bit problem? Did you work with the 32-Bit version?

Yes.

> So, I think this is not a problem that is influenced by the selection of the output buffer size.

I'm not sure why this occurs either, but you may want to try changing the VST priority to Very High in task manager.  Regardless, I find no evidence to support this VSTs playback issue is specific to LMMS as it was reproducible on my machine (Windows XP SP3) without LMMS running.

**Comment by oeai** [comment]
in DSP pathcmix there's a setup of output 48\96 hz, is the output goes
through this sound card?
if lmms uses 44.1 then there should be set 44khz as well.
there's also 24\32 bit setup for cards, i got scratchy sound for optical
io if those bits are different for input and output.
maybe i'll try to reproduce.

On 10.11.2015 04:47, Tres Finocchiaro wrote:
>
>     I am using the 32-Bit version of LMMS and ProteusVX. Possibly, it
>     is a 32-Bit problem? Did you work with the 32-Bit version?
>
> Yes.
>
>     So, I think this is not a problem that is influenced by the
>     selection of the output buffer size.
>
> I'm not sure why this occurs either, but you may want to try changing
> the VST priority to Very High in task manager. Regardless, I find no
> evidence to support this VSTs playback issue is specific to LMMS as it
> was reproducible on my machine (Windows XP SP3) without LMMS running.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LMMS/lmms/issues/2440#issuecomment-155256917>.
>


-- 
OEAI

**Comment by oeai** [comment]
![proteus](https://cloud.githubusercontent.com/assets/2557665/11064635/76a47ac2-87cf-11e5-9538-f3e9c6755469.png)
i got it working through vestige. check the session settings near FX button in DSP, also check windows settings for your cards.
